ddtrace/llmobs/_integrations/langchain.py

```diff
@@ -44,43 +44,52 @@ def llmobs_set_tags(
         span.set_tag_str(MODEL_NAME, span.get_tag(MODEL) or "")
         span.set_tag_str(MODEL_PROVIDER, model_provider or "")
 
-        # input parameters
-        if model_provider:
-            input_parameters = {}
-            temperature = span.get_tag(f"langchain.request.{model_provider}.parameters.temperature")
-            max_tokens = span.get_tag(f"langchain.request.{model_provider}.parameters.max_tokens")
-            if temperature:
-                input_parameters["temperature"] = float(temperature)
-            if max_tokens:
-                input_parameters["max_tokens"] = int(max_tokens)
-            if input_parameters:
-                span.set_tag_str(INPUT_PARAMETERS, json.dumps(input_parameters))
+        self._llmobs_set_input_parameters(span, model_provider)
 
         if operation == "llm":
             self._llmobs_set_meta_tags_from_llm(span, inputs, response, error)
-        elif operation == "chat_model":
+        elif operation == "chat":
             self._llmobs_set_meta_tags_from_chat_model(span, inputs, response, error)
         elif operation == "chain":
-            # TODO to be added as a follow-up PR
             pass
         self._llmobs_set_metrics_tags(span)
 
+    def _llmobs_set_input_parameters(
+        self,
+        span: Span,
+        model_provider: Optional[str],
+    ):
+        if model_provider:
+            input_parameters = {}
+            temperature = (
+                span.get_tag(f"langchain.request.{model_provider}.parameters.temperature")
+                or span.get_tag(f"langchain.request.{model_provider}.request.model_kwargs.temperature")
+                or 0.0  # huggingface
+            )
+            max_tokens = (
+                span.get_tag(f"langchain.request.{model_provider}.parameters.max_tokens")
+                or span.get_tag(f"langchain.request.{model_provider}.parameters.maxTokens")
+                or span.get_tag(f"langchain.request.{model_provider}.parameters.model_kwargs.max_tokens")  # ai21
+                or 0  # huggingface
+            )
+            input_parameters["temperature"] = float(temperature)
+            input_parameters["max_tokens"] = int(max_tokens)
```

Comment:
Can we only specify the temp/max_tokens if it's a provided value? i.e. we don't want to specify default max_tokens = 0 if it's not set as a parameter

Reply:
Good point, we'll have to update the bedrock integration as well (not in this PR)